import/index_java/src/org/vufind/index/ConfigManager.java

```diff
@@ -210,55 +221,59 @@ public Ini loadConfigFile(String filename)
                 retVal.put(key, overrideSection.get(key));
             }
         }
+
+        if (retVal == null) {
+            logger.warn(section + " section missing from " + filename);
+        }
         return retVal;
     }
 
     /**
-     * Get a setting from a VuFind configuration file.
+     * @deprecated
+     */
+    public Map<String, String> getSanitizedConfigSection(String filename, String section)
+    {
+        return getConfigSection(filename, section);
+    }
+
+    /**
+     * Get a setting from a VuFind configuration file and sanitize the value.
      * @param filename configuration file name
      * @param section section name within the file
      * @param setting setting name within the section
      */
     public String getConfigSetting(String filename, String section, String setting)
     {
-        String retVal = null;
-
-        // Grab the ini file.
-        Ini ini = loadConfigFile(filename);
+        String retVal = getRawConfigSetting(filename, section, setting);
+        return retVal == null ? retVal : sanitizeConfigSetting(retVal);
+    }
 
-        // Check to see if we need to worry about an override file:
-        String override = ini.get("Extra_Config", "local_overrides");
-        if (override != null) {
-            Ini overrideIni = loadConfigFile(override);
-            retVal = overrideIni.get(section, setting);
-            if (retVal != null) {
-                return sanitizeConfigSetting(retVal);
-            }
+    /**
+     * Get a setting from a VuFind configuration file.
+     * @param filename configuration file name
+     * @param section section name within the file
+     * @param setting setting name within the section
+     */
+    public String getRawConfigSetting(String filename, String section, String setting)
+    {
+        Map<String, String> sectionMap = getConfigSection(filename, section);
+        if (sectionMap == null) {
+            return null;
         }
 
-        // Try to find the requested setting:
-        retVal = ini.get(section, setting);
-
-        //  No setting?  Check for a parent configuration:
-        while (retVal == null) {
-            String parent = ini.get("Parent_Config", "path");
-            if (parent !=  null) {
-                try {
-                    ini.load(new FileReader(new File(parent)));
-                } catch (Throwable e) {
-                    dieWithError(
-                        "Unable to access " + parent
-                        + " (" + e.getMessage() + ")"
-                    );
-                }
-                retVal = ini.get(section, setting);
-            } else {
-                break;
-            }
+        String retVal = sectionMap.get(setting);
+        if (retVal == null) {
+            logger.warn(section + "." + setting + " setting missing from " + filename);
```

Comment:
Upon reviewing the way getConfigSetting is used in the rest of the code, I don't think this warning is a good idea. There are many places where we request optional settings and act upon null return values, and all of those scenarios will fill the log with warnings. I think if we have logging at all, it should be "opt in" either through an argument or through a separate method (like getRequiredConfigSetting/getRawRequiredConfigSetting). I'd be inclined to just remove this for simplicity, but if you think it's important, we can discuss further! Maybe that's a project for a separate PR, though, since there's already a lot going on in this one.